### ISSUE TITLE
[bug] Chart tooltip is one day back

### ISSUE THREAD

**Issue by epoberezkin** [pvorb/npm-stat.com#43]
See https://npm-stat.com/charts.html?package=ajv
or https://npm-stat.com/charts.html?package=ajv&from=2017-04-01&to=2017-04-16

Particularly on the second one you can see that the date one the axis is correct and in the tooltip (when you mouse-over the dot on the chart) it is one day back

**Comment by pvorb** [comment]
Yep, I know this issue. It has been introduced in PR #37, but I didn't recognize it before deploying that change and didn't feel like it's that important.

I guess it is caused by some timezone problem. Can you tell me in which timezone you tested this behavior?

**Comment by epoberezkin** [comment]
In GMT (UTC +1 now)

**Comment by epoberezkin** [comment]
looks fixed

**Comment by pvorb** [comment]
Are you still in UTC+1? Because I haven't changed the calculation, I think.

**Comment by epoberezkin** [comment]
Maybe it’s because it’s UTC now...

**Comment by pvorb** [comment]
That sounds likely.

**Comment by ozzyogkush** [comment]
I'm seeing this issue and I am in Eastern Time.

**Comment by vlzhr** [comment]
Looks like the bug appears in a chart generation. The plain API stat is [correct](https://npm-stat.com/api/download-counts?package=%40waves%2Fsurfboard&from=2019-07-24&until=2019-09-24) while the chart [lies](https://npm-stat.com/charts.html?package=%40waves%2Fsurfboard&from=2019-07-24&until=2019-09-24)
![image](https://user-images.githubusercontent.com/22869641/65495595-af23f580-debf-11e9-9523-607745998442.png)
![image](https://user-images.githubusercontent.com/22869641/65495671-cfec4b00-debf-11e9-80f8-c22238779001.png)

**Comment by pvorb** [comment]
@ozzyogkush 

> IMHO it should give data for the current day even if the day is not over.

npm-stat.com cannot give numbers before a day is over, since NPM starts calculating the numbers on midnight UTC.

@vlzhr Yes, everything is fine in the back end, but time zones in JavaScript are awful. Highcharts uses `Date` objects for displaying the x-axis and 2019-09-23T00:00:00Z might be displayed as Sunday, Sep 22, 2019 if your offset from UTC is negative.

**Comment by seam345** [comment]
looks like this is linked to the issue I just opened #83 where I noticed the timezone moving forwards has caused it to duplicate a date and going back has caused it to skip a date. This results in the graph appearing as above.
